### ISSUE TITLE
evaluate requires to define optimizer in compile of restored model

### ISSUE THREAD

**Issue by christian-rauch** [keras-team/keras#8501]
Following up from https://github.com/fchollet/keras/issues/2621:

I want to evaluate a combination of models restored from trained weights. `compile` requires me to define `optimizer` and `loss` which should only be required during training.

E.g. I am restoring two models from hdf5 files and connect the output of the first to the input of the second:
```
m1 = load_model(file1)
m2 = load_model(file2)
m = Model(inputs=m1.input, outputs=m2(m1.output))
```

For evaluation, I would expect to only provide an evaluation metric, e.g.:
```
m.compile(metrics="mse")
test_error = m.evaluate_generator(generator=..., steps=...)
```

I think that `optimizer` and `loss` should be optional for `compile` and only be required for training, but that `evaluate` / `evaluate_generator` should require `metrics` to be defined.
